src/main/scala/scorex/core/network/SendingStrategy.scala

```diff
@@ -7,11 +7,16 @@ trait SendingStrategy {
 }
 
 object SendToRandom extends SendingStrategy {
+
+  private val limit: Long = 3 * 60 * 60 * 1000 // 3h
+
   override def choose(peers: Seq[ConnectedPeer]): Seq[ConnectedPeer] = {
-    if (peers.nonEmpty) {
-      Seq(peers(Random.nextInt(peers.length)))
+    val activeRecently: Seq[ConnectedPeer] = // choose peers which were active in the last limit hours
+      peers.filter(System.currentTimeMillis - _.lastMessage  < limit)
```

Comment:
this filter is saying that the node is not sending messages to peers not active in last 3 hours. However, they will be disconnected before, so the code is not needed. 

The issue was about filtering peers sent in response to GetPeer network message. They are chosen is SeenPeers:

```
    case class SeenPeers(howMany: Int) extends GetPeers[Seq[PeerInfo]] {

      override def choose(knownPeers: Map[InetSocketAddress, PeerInfo],
                          blacklistedPeers: Seq[InetAddress],
                          sc: ScorexContext): Seq[PeerInfo] = {
        val recentlySeenNonBlacklisted = knownPeers.values.toSeq
          .filter { p =>
            (p.connectionType.isDefined || p.lastHandshake > 0) &&
              !blacklistedPeers.exists(ip => p.peerSpec.declaredAddress.exists(_.getAddress == ip))
          }
        Random.shuffle(recentlySeenNonBlacklisted).take(howMany)
      }
    }
```

so you need to add a filter to this code (and write a test).

Reply:
Should I use lastHandshake or lastMessage to filter peers?

Reply:
done